doc/neps/nep-0052-python-api-cleanup.rst

```diff
@@ -37,33 +38,55 @@ NumPy has a very large, and not very well-defined, API surface:
+been hidden from ``__dir__``.
+A particularly problematic example of that is ``np.core``,
+which is technically private but heavily used in practice.
+For a full overview of what's consider public, private or a bit in between, see
+`<https://github.com/numpy/numpy/blob/main/numpy/tests/test_public_api.py>`__.
 
-The size and lack of clear boundaries of this API surface has costs:
+The size of the API and the lacking definition of its boundaries
+incur significant costs:
 
-- For users it is difficult to choose between similar functions or know what
-  the recommended way of implementing something is. Looking for functions with
-  tab completion in IPython or a notebook or IDE is a challenge (e.g., type
-  ``np.<TAB>`` and look at the first six items offered - there's two ufuncs
+- **Users find it hard to disambiguate between similarly named
+  functions.**
+
+  Looking for functions with
+  tab completion in IPython, notebook, or IDE is a challenge (e.g., type
+  ``np.<TAB>`` and look at the first six items offered: two ufuncs
   (``abs``, ``add``), one alias (``absolute``), and three functions that are
-  not useful to them (``add_docstring``, ``add_newdoc``, ``add_newdoc_ufunc``).
+  not intended for end-users (``add_docstring``, ``add_newdoc``, ``add_newdoc_ufunc``).
   As a result, the learning curve for NumPy is steeper than it has to be.
-- For maintainers of other libraries which aim to implement a NumPy-compatible
-  API - Dask, CuPy, JAX, PyTorch, TensorFlow, cuNumeric, etc. - or that aim to
-  support NumPy in some other way - Numba, Pythran - there is an implementation
-  cost to every extra object in the namespace. In practice, no other library
-  has full support for everything in NumPy, and making decisions about niche or
-  legacy objects in NumPy is a time-consuming effort.
-- Teaching the NumPy API to others or making it accessible in a
-  learner-oriented GUI has similar costs.
+
+- **Libraries that mimic the NumPy API face significant implementation barriers.**
+
+  For maintainers of NumPy API-compatible array libraries (Dask, CuPy, JAX, PyTorch,
+  TensorFlow, cuNumeric, etc.) and transpilers (Numba, Pythran, Cython, etc.) there is an implementation
+  cost to each object in the namespace. In practice, no other library
+  has full support for the entire NumPy, partly because it is so hard
+  to know what to include faced with a slew of aliases and legacy
+  objects.
+
+- **Teaching NumPy is more complicated than it needs to be.**
+
+  Similarly, a larger API is confusing to learners, who not only have
+  to *find* functions but have to choose *which* to use.
+
+- **Developers are hesitant to grow the API surface.**
+
+  This happens even when the changes are warranted, because they are
+  aware of the above concerns.
 
 Link discussion about restructuring namespaces! (e.g., find the thread with the
 GUI explorer person)
 
+.. S: I first thought you were talking about Manim,
+   but looks like it's something different.
```

Comment:
Yeah I can't find the name anymore - it was in response to a discussion about starting to organize `np.lib` better.